modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java

```diff
@@ -185,11 +200,40 @@ private void ping(WebSocketSession ws) {
      * @throws IOException If failed to process.
      */
     private void registerAgent(WebSocketEvent evt, WebSocketSession ws) throws IOException {
```

Comment:
Why you pass WebSocketSession, WebSocketEvent? Code in this method should only validate and construct response object.

Reply:
This allows to make "handleAgentEvents()" method simple.
"registerAgent()" - is a private method, I like have deserialization of AgentHandshakeRequest in this method for now.

Reply:
But now your business logic depends from socket implementation, have many nested `if-else`. Look at rest-http module we don't pass Servlets specific code into `CommandHandlers`